cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -931,72 +964,122 @@ public void testCrossNSDatasetAccessWithAuthMapReduce() throws Exception {
     Map<String, String> argsForMR = ImmutableMap.of(DatasetCrossNSAccessWithMAPApp.INPUT_KEY, "table1",
                                                     DatasetCrossNSAccessWithMAPApp.OUTPUT_KEY, "table2");
     MapReduceManager mrManager = appManager.getMapReduceManager(DatasetCrossNSAccessWithMAPApp.MAPREDUCE_PROGRAM);
-    // Switch to Bob. Should fail as Bob do not have access to the dataset namespace
+
+    // give BOB ALL permission in the
+    grantAndAssertSuccess(AUTH_NAMESPACE, BOB, EnumSet.of(Action.ALL));
+
+    // Switch to BOB and run the  mapreduce job. The job will fail at the runtime since BOB does not have permission
+    // on the input and output datasets in another namespaces.
     SecurityRequestContext.setUserId(BOB.getName());
-    try {
-      mrManager.start(argsForMR);
-      mrManager.waitForFinish(5, TimeUnit.MINUTES);
-      Assert.fail("Should fail here since bob does not have the privileges on input dataspace.");
-    } catch (Exception e) {
-      // Expected
-    }
+    verifyProgramFailure(argsForMR, mrManager);
```

Comment:
In Flowlets we are testing for its status goes into running and then call stop on it which will make the status of run killed since its a realtime program unlike mapreduce and spark where the program stops  when the runtime exception happens and the status is marked as failed.